### ISSUE TITLE
Plugin host logging doesn't respect application log level

### ISSUE THREAD

**Issue by colin-grant-work** [eclipse-theia/theia#12234]
### Feature Description:

When starting the plugin host process, listeners are added to its `stdout` and `stderr` channels to produce logs on the main application's backend log channel:

https://github.com/eclipse-theia/theia/blob/e2f76b6b119749864bf8d93bcfa07cb59f522059/packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts#L204-L206

However, inside the host process, the default node console logger is used, so there is no distinction between debug and info level logging or warning and error level logging; both of the former go to `stdout` and both of latter go to `stderr`. That means that a change like [this one](https://github.com/eclipse-theia/theia/pull/12224/files#diff-973b848c452142d9c36766d4ca9905f2dab43b5e4cabb6068560b633635e4372) actually has no effect, because the host process will treat debug and log levels as the same. It would be nice to allow the same distinctions on log levels in the hosted process as in the main process.

**Comment by tortmayr** [comment]
I agree, this would definitely be helpful to reduce the default logging noise of the plugin host.
I guess we could extend the  [`hosted-plugin-protocol`](https://github.com/eclipse-theia/theia/blob/e2f76b6b119749864bf8d93bcfa07cb59f522059/packages/plugin-ext/src/hosted/node/hosted-plugin-protocol.ts) and send the log message + level over the process's IPC channel and then forward to the corresponding `logger` method.
We then also have to rebind the default `console` logging method in the [plugin-host](https://github.com/eclipse-theia/theia/blob/e2f76b6b119749864bf8d93bcfa07cb59f522059/packages/plugin-ext/src/hosted/node/plugin-host.ts) similar to how we do it in the [`Logger` API](https://github.com/eclipse-theia/theia/blob/e2f76b6b119749864bf8d93bcfa07cb59f522059/packages/core/src/common/logger.ts#L42).
